Scripts/Generation_scripts/Grid.py

```diff
@@ -18,13 +18,33 @@
 import random
 
 class Grid():
-    def __init__(self):
-        self.location = 'Bahraich'
-        self.CLOVER_filepath = '/***YOUR LOCAL FILE PATH***/CLOVER 4.0'
+    def __init__(self,**kwargs):
+        # kwargs must include location
+        # kwargs can include grid_inputs (to override inputs in location files)
+        self.kwargs = kwargs
```

Comment:
In general, leaving things in the `kwargs` member and storing it on the class isn't that great, as people then don't know what's there. E.G., later on, the `grid_inputs` keyword argument is accessed, but it's not clear that this was a keyword argument used to instantiate the class. Better would be to store the `grid_inputs` on the class explicitly:
```
def __init__(self, **kwargs):
    ...
    self.grid_inputs = kwargs.get("grid_inputs")
```